android/cli/lib/android-manifest.js

```diff
@@ -598,20 +598,30 @@ class AndroidManifest {
 			throw new Error('1st argument must be of type string.');
 		}
 
-		// Inject commonly used Android XML namespaces into the <manifest/> element if missing.
-		// Note: Needed for backward compatibility since older versions of Titanium would inject them too.
+		// Process the XML string before parsing it.
 		const MANIFEST_ELEMENT_PREFIX = '<manifest';
 		const manifestIndex = text.indexOf(MANIFEST_ELEMENT_PREFIX);
 		if (manifestIndex >= 0) {
+			// Inject commonly used Android XML namespaces into the <manifest/> element if missing.
+			// Needed for backward compatibility since older versions of Titanium would inject them too.
 			const insertIndex = manifestIndex + MANIFEST_ELEMENT_PREFIX.length;
-			if (text.indexOf('xmlns:tools') < 0) {
+			let attributeIndex = text.indexOf('xmlns:tools', manifestIndex);
+			if ((attributeIndex < 0) || (attributeIndex > text.indexOf('>', manifestIndex))) {
 				const attributeString = ' xmlns:tools="http://schemas.android.com/tools"';
 				text = text.substring(0, insertIndex) + attributeString + text.substring(insertIndex);
 			}
-			if (text.indexOf('xmlns:android') < 0) {
+			attributeIndex = text.indexOf('xmlns:android', manifestIndex);
+			if ((attributeIndex < 0) || (attributeIndex > text.indexOf('>', manifestIndex))) {
 				const attributeString = ' xmlns:android="http://schemas.android.com/apk/res/android"';
 				text = text.substring(0, insertIndex) + attributeString + text.substring(insertIndex);
 			}
+
+			// Remove "xmlns:android" namespace attributes from all child elements. Only supported in <manifest/>.
+			// Note: CLI used to wrongly inject these after updating "tiapp.xml" if missing from root element.
+			const manifestEndIndex = text.indexOf('>', manifestIndex);
+			text = text.replace(/xmlns:android=".*?"/g, (match, p1) => {
```

Comment:
In the regex `/xmlns:android=".*?"/g` is the `?` necessary, wouldn't the `.*` suffice?

Reply:
Yes, the `?` is needed. Without it, the regex goes past the attribute and to the last double quote before the next match.